pkg/process/config/config_test.go

```diff
@@ -115,29 +115,41 @@ func TestBlacklist(t *testing.T) {
 // TestEnvGrpcConnectionTimeoutSecs tests DD_PROCESS_CONFIG_GRPC_CONNECTION_TIMEOUT_SECS.
 // This environment variable cannot be tested with the other environment variables because it is overridden.
 func TestEnvGrpcConnectionTimeoutSecs(t *testing.T) {
-	syscfg, err := sysconfig.Merge("")
+	oldConfig := config.Datadog
+	t.Cleanup(func() { config.Datadog = oldConfig })
+
+	syscfg, err := sysconfig.New("")
 	require.NoError(t, err)
 
 	t.Run("DD_PROCESS_CONFIG_GRPC_CONNECTION_TIMEOUT_SECS", func(t *testing.T) {
+		config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+		config.InitConfig(config.Datadog)
 		t.Setenv("DD_PROCESS_CONFIG_GRPC_CONNECTION_TIMEOUT_SECS", "1")
 		_, _ = NewAgentConfig("test", "", syscfg)
 		assert.Equal(t, 1, config.Datadog.GetInt("process_config.grpc_connection_timeout_secs"))
 	})
 
 	t.Run("DD_PROCESS_AGENT_GRPC_CONNECTION_TIMEOUT_SECS", func(t *testing.T) {
+		config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+		config.InitConfig(config.Datadog)
 		t.Setenv("DD_PROCESS_AGENT_GRPC_CONNECTION_TIMEOUT_SECS", "2")
 		_, _ = NewAgentConfig("test", "", syscfg)
 		assert.Equal(t, 2, config.Datadog.GetInt("process_config.grpc_connection_timeout_secs"))
 	})
 }
 
 func TestYamlConfig(t *testing.T) {
+	oldConfig := config.Datadog
+	t.Cleanup(func() {
+		config.Datadog = oldConfig
+	})
 	// Reset the config
 	config.Datadog = config.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+	config.InitConfig(config.Datadog)
 
-	f, err := ioutil.TempFile("", "yamlConfigTest*.yaml")
-	defer os.Remove(f.Name())
-	assert.NoError(t, err)
+	f, err := os.CreateTemp(t.TempDir(), "yamlConfigTest*.yaml")
+	require.NoError(t, err)
+	defer f.Close()
```

Comment:
Mind clarifying this change?

Reply:
`ioutil.TempFile` is considered deprecated as of 1.17. It is calling `os.CreateTemp` underneath. Using `t.TempDir()` ensures that the temp file is automatically cleaned up when the test completes.

Reply:
Ah, nice, thank you.